substrate/primitives/core/src/ecdsa.rs

```diff
@@ -569,6 +577,7 @@ mod test {
 	}
 
 	#[test]
+	#[cfg(feature = "std")]
```

Comment:
Yes. I want `time cargo test --release -p sp-core --no-default-features --features="serde"` to be executed correctly.

Reply:
But you don't run this anymore? And serde is mainly about testing that it compiles?

Reply:
I want to run it locally. I want to have ability to test the `sp-core` with the different feature sets it offers. `serde` was just an example, `full-crypto` or `full-crypto,serde` are the other options.

Reply:
We have trait methods definition under `std`, so w/o this paricular gate you will get:

`cargo +nightly check --tests --no-default-features --features="full_crypto"`

```
error[E0599]: no function or associated item named `from_full` found for struct `ecdsa::Public` in the current scope
   --> substrate/primitives/core/src/ecdsa.rs:588:12
    |
80  | pub struct Public(pub [u8; PUBLIC_KEY_SERIALIZED_SIZE]);
    | ----------------- function or associated item `from_full` not found for this struct
...
588 |             Public::from_full(
    |                     ^^^^^^^^^ function or associated item not found in `Public`

For more information about this error, try `rustc --explain E0599`.
error: could not compile `sp-core` (lib test) due to previous error
```

Reply:
> We have trait methods definition under `std`, so w/o this paricular gate you will get:
> 
> `cargo +nightly check --tests --no-default-features --features="full_crypto"`

Because you are using `--tests`. Just drop this, these tests only need to work on `std`.

Reply:
Then this PR does not make sense.

Reply:
I mean checking that it compiles make sense. (without the tests). To ensure that these features compile on their own.